pkg/controller/directimagemigration/validate.go

```diff
@@ -0,0 +1,273 @@
+			Category: Critical,
+			Message:  InvalidSourceClusterReferenceMessage,
+		})
+		return nil
+	}
+
+	// Not ready
+	if !cluster.Status.IsReady() {
+		imageMigration.Status.SetCondition(migapi.Condition{
+			Type:     SourceClusterNotReady,
+			Status:   True,
+			Category: Critical,
+			Message:  SourceClusterNotReadyMessage,
+		})
+	}
+	// Exposed registry path
+	registryPath, err := cluster.GetRegistryPath(r)
+	if err != nil || registryPath == "" {
+		imageMigration.Status.SetCondition(migapi.Condition{
+			Type:     MissingSourceClusterRegistryPath,
+			Status:   True,
+			Category: Critical,
+			Message:  MissingSourceClusterRegistryPathMessage,
+		})
+	}
+	return nil
+}
+
+func (r ReconcileDirectImageMigration) validateDestCluster(imageMigration *migapi.DirectImageMigration) error {
+	ref := imageMigration.Spec.DestMigClusterRef
+
+	if !migref.RefSet(ref) {
+		imageMigration.Status.SetCondition(migapi.Condition{
+			Type:     InvalidDestinationClusterRef,
+			Status:   True,
+			Reason:   NotSet,
+			Category: Critical,
+			Message:  InvalidDestinationClusterReferenceMessage,
+		})
+		return nil
+	}
+
+	// Check if clusters are unique
+	if reflect.DeepEqual(ref, imageMigration.Spec.SrcMigClusterRef) {
+		imageMigration.Status.SetCondition(migapi.Condition{
+			Type:     InvalidDestinationCluster,
+			Status:   True,
+			Reason:   NotDistinct,
+			Category: Critical,
+			Message:  InvalidDestinationClusterMessage,
+		})
+		return nil
+	}
+
+	cluster, err := migapi.GetCluster(r, ref)
+	if err != nil {
+		return liberr.Wrap(err)
+	}
+
+	// Not found
+	if cluster == nil {
+		imageMigration.Status.SetCondition(migapi.Condition{
+			Type:     InvalidDestinationClusterRef,
+			Status:   True,
+			Reason:   NotFound,
+			Category: Critical,
+			Message:  InvalidDestinationClusterReferenceMessage,
+		})
+		return nil
+	}
+
+	// Not ready
+	if !cluster.Status.IsReady() {
+		imageMigration.Status.SetCondition(migapi.Condition{
+			Type:     DestinationClusterNotReady,
+			Status:   True,
+			Category: Critical,
+			Message:  DestinationClusterNotReadyMessage,
+		})
+	}
+	// Exposed registry path
+	registryPath, err := cluster.GetRegistryPath(r)
+	if err != nil || registryPath == "" {
+		imageMigration.Status.SetCondition(migapi.Condition{
+			Type:     MissingDestinationClusterRegistryPath,
+			Status:   True,
+			Category: Critical,
+			Message:  MissingDestinationClusterRegistryPathMessage,
+		})
+	}
+	return nil
+}
+
+// Validate required namespaces on the source cluster.
+// Returns error and the total error conditions set.
+func (r ReconcileDirectImageMigration) validateNamespaces(imageMigration *migapi.DirectImageMigration) error {
+	count := len(imageMigration.Spec.Namespaces)
+	if count == 0 {
+		imageMigration.Status.SetCondition(migapi.Condition{
+			Type:     NsListEmpty,
+			Status:   True,
+			Category: Critical,
+			Message:  NsListEmptyMessage,
+		})
+		return nil
+	}
+	cluster, err := imageMigration.GetSourceCluster(r)
```

Comment:
rename to `srcCluster` ?

Reply:
I could do that. We've generally been just using cluster/client when a function only references one, but maybe it's clearer to spell it out here as well.

Reply:
yeah, I have been tripped on reading that code, always need to backtrack to where the variable is defined to understand what client the call is made against. Would be nice to have this explicitness